### PR TITLE
Fix carry over logic for fractional formatting

### PR DESCRIPTION
**Pull request by hl662**
Addresses a bug in the fractional formatting logic for composite units in the `@itwin/core-quantity` package. This ensures that rounding of fractional components correctly carries over to the parent unit when appropriate, preventing formatting errors in edge cases. Comprehensive tests have been added to verify the new behavior for both fractional and decimal formats.

**Bug fix in fractional formatting logic:**

* Fixed the carry-over logic in `Formatter.formatQuantity` so that when the fractional part of a unit rounds up to a full unit in the next smaller component (e.g., inches rounding up to feet), the parent unit is incremented appropriately. This prevents cases like "11.9999999 ft" from incorrectly displaying as "11'-12\"" instead of "12'-0\"".

**Testing and validation:**

* Added extensive test cases to `CompositeFormats.test.ts` to verify correct carry-over behavior for fractional formats, including positive, negative, and edge values for both feet/inches and degrees/minutes/seconds.
* Added tests to ensure that decimal formats do not use the fractional carry-over logic, maintaining their expected behavior.

### CODE REVIEW

File: core/quantity/src/Formatter/Formatter.ts [hunk 1]
```diff
@@ -244,7 +244,30 @@ export class Formatter {
...
+        let wholePart = Math.trunc(unitValue);
+
+        // Check if the remaining fractional part will round up to a full unit in the next (smaller) component
+        if (spec.format.type === FormatType.Fractional && i === spec.unitConversions.length - 2) {
+          // For the second-to-last unit with fractional formatting, check if rounding causes carry-over
+          const fractionalPart = unitValue - wholePart;
+          const nextUnitValue = applyConversion(fractionalPart, spec.unitConversions[i + 1].conversion);
+
+          // Create a FractionalNumeric to determine what the rounded value would be
+          const fn = new FractionalNumeric(Math.abs(nextUnitValue), spec.format.precision as FractionalPrecision, true);
```
Comment (ben-polinsky) [inline core/quantity/src/Formatter/Formatter.ts:256]: This cast is jumping out. Why are you sure the precision is FractionalPrecision at this point?

Comment (hl662) [reply core/quantity/src/Formatter/Formatter.ts:256]: 2 things : 
1. By this point in `Formatter`, the instantiated Format has already validated the precision value through [parsePrecision](https://github.com/iTwin/itwinjs-core/blob/nam/qty-rounding-bug/core/quantity/src/Formatter/Format.ts#L150-L151), which validates whether the format type is fractional or non fractional.
2. We check above if the spec format type is `Fractional` on top of the first point as well. This follows the same logic Bill added years ago [in another place he used casting](https://github.com/iTwin/itwinjs-core/blob/nam/qty-rounding-bug/core/quantity/src/Formatter/Formatter.ts#L367)

**Review by ben-polinsky — APPROVED** [review]

**Review by rbreukelaar — APPROVED** [review]
Yes, this looks good, and thorough.
